### PR TITLE
Extend documentation for encryption and signing parameters

### PR DESCRIPTION
**Pull request by lieser**

### CODE REVIEW

**Comment by reneme** [comment]
Nice! This information would have been so helpful for me personally when i started out with Botan. Thanks for the effort!

(Apparently GitHub has some issues with the comments/review UI right now; hence actual review will have to wait.)

File: doc/api_ref/pubkey.rst [hunk 1]
```diff
@@ -585,6 +584,54 @@ the private key.
...
+
+- Names: ``OAEP`` / ``EME-OAEP`` / ``EME1``
+- Parameters specification:
+
+  - ``(<HashFunction>)``
+  - ``(<HashFunction>,MGF1)``
+  - ``(<HashFunction>,MGF1(<HashFunction>))``
+  - ``(<HashFunction>,MGF1(<HashFunction>),<optional label>)``
+
+- The only Mask generation function available is MGF1 and the default.
```
Comment (reneme) [inline doc/api_ref/pubkey.rst:611]:
```suggestion
- The only Mask generation function available is MGF1 which is also the default.
```

**Review by reneme — APPROVED** [review]

File: doc/api_ref/pubkey.rst (hunk 1 above)
Comment (lieser) [reply doc/api_ref/pubkey.rst]: changed

File: doc/api_ref/pubkey.rst [hunk 2]
```diff
@@ -708,15 +755,27 @@ Signatures are verified using
...
+   Takes one of the following as parameter:
+
+   - ``<user ID>`` (uses ``SM3``)
+   - ``<user ID>,<HashFunction>``
+
+#. Dilithium.
+   Takes the optional parameter ``Deterministic`` (default) or ``Randomized``.
+#. SPHINCS+.
+   Takes the optional parameter ``Deterministic`` (default) or ``Randomized``.
+#. XMSS. Takes no parameter.
```
[the remark this answers is not in the document]
Comment (lieser) [reply doc/api_ref/pubkey.rst:780]: I will update it after this is meged

**Pull request merged**